techmatters/flex-plugins
# Create multi-option select field type

Primary reviewer: @stephenhand 

## Description
This PR adds a new type of form input:
```
type ListboxMultiselectDefinition = {
  type: 'listbox-multiselect';
  options: InputOption[];  // the list of options, using the same notation as we've been using so far ( `{ value: x, label: x }` )
  height?: number;  // optionally, pass in the number of pixels that this component should have vertically.
  width?: number;  // optionally, pass in the number of pixels that this component should have horizontally.
} & ItemBase;
```

Example of definition:
```
{
    "name": "actionTaken",
    "label": "Action Taken",
    "type": "listbox-multiselect",
    "options": [
      { "value": "Direct interventions by the child helpline", "label": "Direct interventions by the child helpline" },
      { "value": "Recommendations of resources", "label": "Recommendations of resources" },
      { "value": "Referrals to child protection agencies", "label": "Referrals to child protection agencies" },
      { "value": "Referrals to general healthcare professionals", "label": "Referrals to general healthcare services" },
      { "value": "Referrals to law enforcement agencies", "label": "Referrals to law enforcement agencies" },
      { "value": "Referrals to mental health services", "label": "Referrals to mental health services" },
      { "value": "Referrals to other organisations", "label": "Referrals to other organisations" },
      { "value": "Referrals to school counsellors", "label": "Referrals to school counsellors" },
      { "value": "Reports to Child Sexual Abuse Material", "label": "Reports to Child Sexual Abuse Material" },
      { "value": "Other", "label": "Other" },
    ],
    "required": { "value": true, "message": "RequiredFieldError" }
  }
```
Above definition results in

![Screenshot from 2022-05-06 11-24-07](https://user-images.githubusercontent.com/15805319/167152599-d1687d31-ccdf-4a75-bf8b-b2118908ba62.png)

As this is an array value, `maxLength` and `minLength` can be passed in to the definition to specify the minimum or the maximum amount of selected options to be valid.

### Checklist
- [N/A] Tested for chat contacts
- [N/A] Tested for call contacts

### Verification steps
- Add the example definition to one or more form definition.
- `npm ci` & `npm run dev` to start the local server.
- Play around with the new input to test appearance and behavior, check that it's saved as an array in HRM and it's properly retrieved in the forms consumers (contact details, case forms).


WIP:
- tests
